spec/rubocop/cop/lint/redundant_with_object_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Lint::RedundantWithObject do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  it 'registers an offense when using `ary.each_with_object { |v| v }`' do
+    expect_offense(<<-RUBY.strip_indent)
+      ary.each_with_object([]) { |v| v }
+          ^^^^^^^^^^^^^^^^^^^^ Use `each` instead of `each_with_object`.
+    RUBY
+  end
+
+  it 'registers an offense when using `ary.each.with_object([]) { |v| v }`' do
+    expect_offense(<<-RUBY.strip_indent)
+      ary.each.with_object([]) { |v| v }
+               ^^^^^^^^^^^^^^^ Remove redundant `with_object`.
+    RUBY
+  end
+
+  it 'autocorrects to ary.each from ary.each_with_object([])' do
+    new_source = autocorrect_source('ary.each_with_object([]) { |v| v }')
+
+    expect(new_source).to eq 'ary.each { |v| v }'
+  end
+
+  it 'autocorrects to ary.each from ary.each.with_object([]) { |v| v }' do
+    new_source = autocorrect_source('ary.each.with_object([]) { |v| v }')
+
+    expect(new_source).to eq 'ary.each { |v| v }'
+  end
```

Comment:
Could you add tests for without parens and/or do-end keyword block.

For example:

```ruby
a.each_with_object([]) do |v|
end
a.each_with_object [] do |v|
end
a.each_with_object([]) {|v| v}
a.each_with_object [] {|v| v}
```


And the auto-correction for the first case is broken.

Reply:
Oh, I didn't realize that. I'm going to see in detail later. Thanks for your advise!

Reply:
I added some test cases and fixed it with 95b0c46.